src/main/java/doit/shop/controller/user/service/UserService.java

```diff
@@ -0,0 +1,63 @@
+package doit.shop.controller.user.service;
+
+import doit.shop.controller.user.domain.User;
+import doit.shop.controller.user.domain.UserRepository;
+import doit.shop.controller.user.dto.UserInfoResponse;
+import doit.shop.controller.user.dto.UserLoginRequest;
+import doit.shop.controller.user.dto.UserLoginResponse;
+import doit.shop.controller.user.dto.UserSignUpRequest;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestParam;
+
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class UserService {
+
+    private final UserRepository userRepository;
+
+    public void checkDuplicatedId(String userLoginId){
+        User user = userRepository.findByUserLoginId(userLoginId);
+        if(user != null){
+            throw new IllegalArgumentException("이미 존재하는 아이디입니다.");
+        }
+    }
+
+    public UserLoginResponse signUp(UserSignUpRequest userSignUpRequest){
+        if(userRepository.findByUserLoginId(userSignUpRequest.userLoginId()) != null){
+            throw new IllegalArgumentException("이미 존재하는 아이디입니다.");
+        }
+
+        User user = User.builder()
```

Comment:
아래처럼 빌더 내용을 도메인 쪽에서 처리해주시면 서비스 코드가 간결해지고 좋을 것 같습니다!

```suggestion
        User user = User.from(userSignupRequest);
```